From: David Weinehall <tao@debian.org>

This patch, should you choose to accept it, makes cron-apt a
/bin/sh-script instead of a /bin/bash-script.


Regards: David Weinehall


diff -ur cron-apt-0.0.20-old/src/cron-apt cron-apt-0.0.20/src/cron-apt
--- cron-apt-0.0.20-old/src/cron-apt    2004-04-16 13:11:20.000000000 +0300
+++ cron-apt-0.0.20/src/cron-apt        2004-07-18 11:28:01.000000000 +0300
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # DocumentId: $Id: better-sh-compatibility.patch 2622 2009-03-24 19:59:59Z ola $
 # Changes:
 #      2003-07-10 Ola Lundqvist <opal@debian.org>
@@ -118,0 +118,0 @@

 ############################### check #########################################

-if [ -e $ERROR -a "$DONTRUN" = "error" ] ; then
+if [ -e $ERROR ] && [ "$DONTRUN" = "error" ] ; then
     echo "Error in last update. Fix the problem and remove $ERROR."
     exit
 fi
@@ -157,7 +157,7 @@
                            cat $ERRORMSGDIR/$ACTIONF >> $ERROR
                        fi
                        cat $TEMP >> $ERROR
-                       if [ "$MAILON" = "error" -o "$MAILON" = "upgrade" -o "$MAILON" = "always" ] ; then
+                       if [ "$MAILON" = "error" ] || [ "$MAILON" = "upgrade" ] || [ "$MAILON" = "always" ] ; then
                            createinfo $MAIL $ACTIONF
                            if [ -f $MAILMSGDIR/$ACTIONF ] ; then
                                cat $MAILMSGDIR/$ACTIONF >> $MAIL
@@ -172,7 +172,7 @@
                                cat $TEMP >> $MAIL
                            fi
                        fi
-                       if [ "$DEBUG" = "error" -o "$MAILON" = "upgrade" -o "$DEBUG" = "verbose" ] ; then
+                       if [ "$DEBUG" = "error" ] || [ "$MAILON" = "upgrade" ] || [ "$DEBUG" = "verbose" ] ; then
                            createinfo $LOG $ACTIONF
                            if [ -f $LOGMSGDIR/$ACTIONF ] ; then
                                cat $LOGMSGDIR/$ACTIONF >> $LOG
